src/material/schematics/ng-generate/mdc-migration/rules/style-migrator.ts

```diff
@@ -89,6 +97,30 @@ export abstract class StyleMigrator {
     atRule.remove();
   }
 
+  /**
+   * Create the new mixin nodes and add them after the provided
+   * all-components-theme mixin node
+   *
+   * @param namespace the namespace being used for angular/material.
+   * @param allComponentThemesNode a all-components-theme mixin node
+   */
+  addNewMixinsAfterNode(namespace: string, allComponentThemesNode: postcss.AtRule) {
+    // We know there will be a theme name since it is an at-include rule
+    const themeName = allComponentThemesNode.params.match(THEME_NAME_REGEX)![0];
+    // Only add the new mixin node if not already replaced for this theme to
+    // avoid duplicate imports
+    if (!this.legacyMixinReplacementThemes.includes(themeName)) {
```

Comment:
Why do we need this check to avoid duplicate imports? Shouldn't this be getting called for each at-include of the all-components-theme mixin?